### ISSUE TITLE
An incorrect item was selected from the search results

### ISSUE THREAD

**Issue by santhosh-g-tz** [maheshj01/searchfield#159]
**Describe the bug**
An incorrect item was selected from the search results

[v1.0.7] (https://pub.dev/packages/searchfield/changelog)

**Screenshots**

https://github.com/user-attachments/assets/f916c54a-fbde-40bc-990e-d51d6f2f1094

**Code sample**
<details>
<summary>Show code sample</summary>

```dart  
var countries = [
      'United States',
      'Germany',
      'Canada',
      'United Kingdom',
      'France',
      'Italy',
      'Spain',
      'Australia',
      'India',
      'China',
      'Japan',
      'Brazil',
      'South Africa',
      'Mexico',
      'Argentina',
      'Russia',
      'Indonesia',
      'Turkey',
      'Saudi Arabia',
      'Nigeria',
      'Egypt',
    ];

   return SearchField<String>(
      maxSuggestionsInViewPort:10,
      suggestions: countries
          .map(
            (e) => SearchFieldListItem<String>(
          e,
          item: e,
          child: Padding(
            padding: const EdgeInsets.all(8.0),
            child: Text(e),
          ),
        ),
      ).toList(),
    );```
</details>

**Comment by maheshj01** [comment]
Thanks for filing that issue, this shows we have a poor tests setup. I will try to get this fixed asap.

**Comment by maheshj01** [comment]
Hi @santhosh-g-tz  Can you please give this a try

add to `pubspec.yaml`

```dart
searchfield:
    git:
      url: git://github.com/maheshj01/searchfield.git
      ref: hotfix-159
```

**Comment by maheshj01** [comment]
@santhosh-g-tz released a new version v1.0.8
